sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -155,7 +155,24 @@ private[hive] class HiveClientImpl(
     }
   }
 
-  ShutdownHookManager.addShutdownHook(() => state.close())
+  private def closeState(): Unit = {
+    // these temp files are registered in o.a.h.u.ShutdownHookManager too
+    // the state.close() will delete them if they are not null and try remove them
+    // from the o.a.h.u.ShutdownHookManager which causes IllegalStateException
+    // we delete them ahead with a high priority hook here and set them to null to
+    // bypass the deletion in state.close()
```

Comment:
Capitalizes the first letter of the first word in a sentence?

Reply:
OK~